src/utils/thread_helper.cpp

```diff
@@ -642,7 +646,11 @@ void ThreadHelper::MgmtSetResponseHandler(otError aResult)
 
     LogOpenThreadResult("MgmtSetResponseHandler()", aResult);
 
-    assert(mAttachHandler != nullptr);
+    if (mAttachHandler == nullptr)
```

Comment:
Should we clear the `mAttachPendingDatasetTlvs`  and `mAttachDelayMs ` when the `mAttachHandler ` is `nullptr`?

Reply:
SGTM. Updated.